### DISCUSSION TITLE
How can we view training data results?

### DISCUSSION THREAD

**Discussion by fathollahzadeh** [openml/automlbenchmark#621] [Q&A]
Hello, 
Based on my current understanding, the system only reports test results. Is there an option to view the training data results as well?

**Comment by PGijsbers** [comment]
This might not be _exactly_ what you are looking for, but some frameworks support a `_save_artifacts` parameter from the framework configuration. See here an example for [autogluon](https://github.com/openml/automlbenchmark/blob/d5a90a8e5e53ba3907d7054585b26a71cda52092/resources/frameworks_2023Q2.yaml#L11) and [h2o](https://github.com/openml/automlbenchmark/blob/d5a90a8e5e53ba3907d7054585b26a71cda52092/resources/frameworks_2023Q2.yaml#L62). Whether a framework supports this depends on their integration, and which items can be saved also depends on the framework. So some inspection of the framework's `exec.py` is necessary. Typically you will find something like a [save_artifacts](https://github.com/openml/automlbenchmark/blob/d5a90a8e5e53ba3907d7054585b26a71cda52092/frameworks/H2OAutoML/exec.py#L169) function if they do support it.

**Reply by fathollahzadeh** [reply]
Thanks for the response.  Having to update the `exec.py ` file for every framework seems inconvenient!
